### PR TITLE
DOC: Fixed example section in pandas/core/dtypes/*.py

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [ ] closes #xxxx
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/dtypes/common.py [hunk 1]
```diff
@@ -1013,11 +1016,7 @@ def is_datetime64_any_dtype(arr_or_dtype) -> bool:
     False
     >>> is_datetime64_any_dtype(np.array([], dtype=np.datetime64))
     True
-    >>> is_datetime64_any_dtype(pd.DatetimeIndex([1, 2, 3],
-                                dtype=np.datetime64))
```
Comment (WillAyd) [inline pandas/core/dtypes/common.py]: I think you can just change dtype here to `"datetime64[ns]"` - not sure on general rules for passing `np.datetime64` without a base as a dtype argument but cc @jbrockmendel

File: pandas/core/dtypes/common.py [hunk 2]
```diff
@@ -1053,14 +1052,9 @@ def is_datetime64_ns_dtype(arr_or_dtype) -> bool:
     False
     >>> is_datetime64_ns_dtype(np.array([], dtype=np.datetime64))  # no unit
     False
-    >>> is_datetime64_ns_dtype(np.array([],
-                               dtype="datetime64[ps]"))  # wrong unit
+    >>> is_datetime64_ns_dtype(np.array([], dtype="datetime64[ps]"))  # wrong unit
     False
-    >>> is_datetime64_ns_dtype(pd.DatetimeIndex([1, 2, 3],
-                               dtype=np.datetime64))  # has 'ns' unit
```
Comment (WillAyd) [inline pandas/core/dtypes/common.py]: same comment here

**Review by WillAyd — CHANGES_REQUESTED** [review]
Thanks for taking a look at these. Is doctest picking these up?

File: pandas/core/dtypes/common.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/dtypes/common.py]: yah, this will raise ValueError.  Either don't pass a dtype at all or pass "datetime64[ns]"

**Comment by jreback** [comment]
@datapythonista @jorisvandenbossche if any comments

**Review by WillAyd — APPROVED** [review]
Lgtm. If there’s a way for doctest to pick these up and automatically check would be all for it in a follow up

File: pandas/core/dtypes/cast.py [hunk 3]
```diff
@@ -1,4 +1,6 @@
-""" routings for casting """
+"""
+Routings for casting.
```
Comment (jorisvandenbossche) [inline pandas/core/dtypes/cast.py]:
```suggestion
Routines for casting.
```

**Review by jorisvandenbossche — CHANGES_REQUESTED** [review]
@MomIsBestFriend can you add those files to the "DOCTESTS" section in code_checks.sh? (or are not yet all docstrings of the files passing?)

**Comment by MomIsBestFriend** [comment]
> @MomIsBestFriend can you add those files to the "DOCTESTS" section in code_checks.sh? (or are not yet all docstrings of the files passing?)

@jorisvandenbossche Yes, all the doctests are now passing, I have added it in 2a330b4

**Review by jreback — APPROVED** [review]
lgtm. @jorisvandenbossche

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Thanks!

**Pull request merged**